### PR TITLE
Fix for bigDecimal values between 0 and 1 having too high of a precision

### PR DESCRIPTION
**Pull request by Jeffery-Wasty**
Changes how precision and scaling are computed for bigDecimal values to account for a big where precision is 1 too high for numbers between 0 and 1.

### CODE REVIEW

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java [hunk 1]
```diff
@@ -1426,6 +1426,32 @@ public void testFailedToResumeTransaction() throws Exception {
             }
         }
 
+        /**
```
Comment (lilgreenbird) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java:1429]: can we test all boundary cases here eg. < 1 and > 1 with precision 38, > 38, etc

Comment (lilgreenbird) [reply src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java:1429]: also, test cstmt as in other tests

File: src/main/java/com/microsoft/sqlserver/jdbc/Parameter.java [hunk 2]
```diff
@@ -531,19 +531,14 @@ private void setTypeDefinition(DTV dtv) {
                             param.typeDefinition = SSType.DECIMAL.toString() + "(" + valueLength + "," + scale + ")";
                         }
                     } else {
-                        BigDecimal bigDecimal = null;
+                        BigDecimal bigDecimal;
                         if (dtv.getJavaType() == JavaType.BIGDECIMAL
                                 && null != (bigDecimal = (BigDecimal) dtv.getSetterValue())) {
```
Comment (lilgreenbird) [inline src/main/java/com/microsoft/sqlserver/jdbc/Parameter.java:536]: since I have comments in other places, can we move this assignment out of the 'if'

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java (hunk 1 above)
Comment (Jeffery-Wasty) [reply src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java:1429]: cstmt introduces rounding to 4 decimal places. For cases with maximum precision, I'm sticking with pstmt, for the cases with less than maximum, I've switched to cstmt.

Test cases include: max precision between 0 and 1, max with number greater than 1, less than max precision for a number between 0 and 1, less than max for a number greater than 1.

File: src/main/java/com/microsoft/sqlserver/jdbc/Parameter.java (hunk 2 above)
Comment (Jeffery-Wasty) [reply src/main/java/com/microsoft/sqlserver/jdbc/Parameter.java]: Moved.

**Review by lilgreenbird — APPROVED** [review]
